test/misc.jl

```diff
@@ -1392,10 +1392,24 @@ end
 @testset "Base/timing.jl" begin
     @test Base.jit_total_bytes() >= 0
 
-    # sanity check `@allocations` returns what we expect in some very simple cases
-    @test (@allocations "a") == 0
-    @test (@allocations "a" * "b") == 0 # constant propagation
-    @test (@allocations "a" * Base.inferencebarrier("b")) == 1
+    # sanity check `@allocations` returns what we expect in some very simple cases.
+    # These are inside functions because `@allocations` uses `Experimental.@force_compile`
+    # so can be affected by other code in the same scope.
+    @test (() -> @allocations "a")() == 0
+    @test (() -> @allocations "a" * "b")() == 0 # constant propagation
+    @test (() -> @allocations "a" * Base.inferencebarrier("b"))() == 1
+
+    lock_conflicts() = parse(Int, read(`$(Base.julia_cmd()) -tauto -E '
+        @lock_conflicts begin
+            l = ReentrantLock()
+            Threads.@threads for i in 1:Threads.nthreads()
+                 lock(l) do
+                    sleep(0.5)
+                end
+            end
+        end
+    '`, String))
+    @test lock_conflicts() > 0 skip=(Sys.CPU_THREADS < 2) # can only test if the worker can multithread
```

Comment:
This is failing on [test x86_64-linux-gnuassertrr](https://buildkite.com/julialang/julia-master/builds/33042#018d69a6-efdb-4b27-ba0d-deb98e5a9967/789-1634)
```
Expression: lock_conflicts() > 0
Evaluated: 0 > 0
```

I guess either `-tauto` isn't setting `nthreads() > 1` or the sleep is too short? But the latter would be surprising?

Reply:
@vtjnash you may see a better test design here?